src/js/components/Pagination/__tests__/Pagination-test.js

```diff
@@ -0,0 +1,262 @@
+import React, { useState } from 'react';
+import 'jest-styled-components';
+import 'jest-axe/extend-expect';
+import 'regenerator-runtime/runtime';
+
+import { cleanup, render } from '@testing-library/react';
+// import { axe } from 'jest-axe';
+import { fireEvent } from '@testing-library/dom';
+
+import { Grommet } from '../../Grommet';
+import { Box } from '../../Box';
+import { Pagination } from '..';
+
+const NUM_ITEMS = 237;
+const STEP = 10;
+const data = [];
+for (let i = 0; i < 95; i += 1) {
+  data.push(`entry-${i}`);
+}
+
+const TestPagination = ({ step, ...rest }) => {
+  const [currentData] = useState(data.slice(0, step));
+
+  return (
+    <>
+      {currentData.map(datum => (
+        <Box key={datum}>{datum}</Box>
+      ))}
+      <Pagination step={step} {...rest} />
+    </>
+  );
+};
+
+describe('Pagination', () => {
+  afterEach(cleanup);
+
+  // test('should have no accessibility violations', async () => {
+  //   const { container } = render(
+  //     <Grommet>
+  //       <Pagination numItems={NUM_ITEMS} />
+  //     </Grommet>,
+  //   );
+
+  //   const results = await axe(container);
+  //   expect(results).toHaveNoViolations();
+  // });
+
+  test(`should display the correct last page based on items length 
+  and step`, () => {
+    const { container, getByText } = render(
+      <Grommet>
+        <Pagination numItems={NUM_ITEMS} />
+      </Grommet>,
+    );
+
+    // default step is 10
+    const expectedPageCount = Math.ceil(NUM_ITEMS / 10);
+    const lastPageButton = getByText(expectedPageCount.toString());
+
+    expect(lastPageButton).toBeTruthy();
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
```

Comment:
How about testing more edge cases? For example, `numItems` in [0, 1, < step, == step, > step].